src/averageAges.js

```diff
@@ -37,7 +48,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let women = people.filter(person => person.sex === 'f');
+
+  if (withChildren) {
+    women = women.filter(woman =>
+      people.some(person => person.mother === woman.name));
+  }
```

Comment:
If the flag is present, then you will run 2 filters. You can do a check inside the filter callback